tuxemon/client.py

```diff
@@ -580,14 +580,14 @@ def push_state(self, state_name: str, **kwargs: Any) -> State:
     @overload
     def push_state(
         self,
-        state_name: Type[StateType],
+        state_name: StateType,
```

Comment:
We should use a `TypeVar` with `StateType` bound here to link the passed state and the returned one, as they are the same.

Reply:
StateType = TypeVar("StateType", bound=State)

Reply:
Already in use.